ipa_room_exploration/ros/src/room_exploration_action_client.cpp

```diff
@@ -63,79 +84,30 @@ int main(int argc, char **argv)
 	}
 	std::cout << "map-size: " << map.rows << "x" << map.cols << std::endl;
 
-//	const std::string topic = "/move_base/global_costmap/costmap";
-//	nav_msgs::OccupancyGrid grid;
-//	grid = *(ros::topic::waitForMessage<nav_msgs::OccupancyGrid>(topic, nh));
-//	ROS_INFO("got grid");
-//
-//	std::vector<signed char> dats;
-//	dats = grid.data;
-//
-//	std::cout << dats.size() << std::endl;
-//	int s = 200;
-//	cv::Mat test_map = cv::Mat(s, s, map.type());
-//
-//	for(size_t u = 0; u < test_map.cols; ++u)
-//	{
-//		for(size_t v = 0; v < test_map.rows; ++v)
-//		{
-//			test_map.at<uchar>(u,v) = (uchar) dats[v+u*s];
-//		}
-//	}
-//
-//	cv::imshow("testtt", test_map);
-//	cv::waitKey();
-
 	ROS_INFO("Waiting for action server to start.");
 	// wait for the action server to start
 	ac.waitForServer(); //will wait for infinite time
 
 	ROS_INFO("Action server started, sending goal.");
 
-	DynamicReconfigureClient drc_exp(nh, "room_exploration_server/set_parameters", "room_exploration_server/parameter_updates");
-	drc_exp.setConfig("room_exploration_algorithm", 2);
-//	drc_exp.setConfig("path_eps", 3);
-//	drc_exp.setConfig("grid_line_length", 15);
-//	drc_exp.setConfig("path_eps", 10);
-//	drc_exp.setConfig("cell_size", 10);
-//	drc_exp.setConfig("plan_for_footprint", true);
-//	drc_exp.setConfig("goal_eps", 0.0);
-//	drc_exp.setConfig("delta_theta", 0.005);
-
-//	cv::Point2f src_center(map.cols/2.0F, map.rows/2.0F);
-//	cv::Mat rot_mat = getRotationMatrix2D(src_center, 180, 1.0);
-//	cv::Mat dst;
-//	cv::warpAffine(map, dst, rot_mat, map.size());
-//	cv::flip(dst, map, 1);
-//	cv::imshow("map", map);
-//	cv::waitKey();
-
 	sensor_msgs::Image labeling;
 	cv_bridge::CvImage cv_image;
 //	cv_image.header.stamp = ros::Time::now();
 	cv_image.encoding = "mono8";
 	cv_image.image = map;
 	cv_image.toImageMsg(labeling);
 
-	geometry_msgs::Pose map_origin;
-	map_origin.position.x = 0.0;
-	map_origin.position.y = 0.0;
+        geometry_msgs::Pose map_origin;
+        map_origin.position.x = origin[0];
+        map_origin.position.y = origin[1];
+        map_origin.position.z = origin[2];
 
 	geometry_msgs::Pose2D starting_position;
-	starting_position.x = 1.0;
-	starting_position.y = 1.0;
-	starting_position.theta = 0.0;
+	starting_position.x = start_pos[0];
+	starting_position.y = start_pos[1];
+	starting_position.theta = start_pos[2];
 
 	std::vector<geometry_msgs::Point32> fov_points(4);
-//	fov_points[0].x = 0.15;		// this field of view fits a Asus Xtion sensor mounted at 0.63m height (camera center) pointing downwards to the ground in a respective angle
```

Comment:
Please keep these lines, as this is the other option to run the example with a field of view planning mode instead of footprint planning. The example file should contain such options because it is explanatory for using the different variants of algorithms of this software. I think it helps new users to see the different options even if commented out.